src/main/java/mil/nga/tiff/FileDirectory.java

```diff
@@ -1274,23 +1274,23 @@ private Number readValue(ByteReader reader, FieldType fieldType) {
 			value = reader.readDouble();
 			break;
 		default:
-			throw new TiffException("Unsupported raster field type: "
-					+ fieldType);
+			throw new TiffException("Unsupported raster sampleType type: "
+					+ sampleType);
 		}
 
 		return value;
 	}
 
 	/**
-	 * Get the field type for the sample
-	 * 
+	 * Get the raster sample type for the sample @see Rasters.SampleType
+	 *
 	 * @param sampleIndex
 	 *            sample index
-	 * @return field type
+	 * @return Sample type
 	 */
-	public FieldType getFieldTypeForSample(int sampleIndex) {
+	public Rasters.SampleType getFieldTypeForSample(int sampleIndex) {
 
-		FieldType fieldType = null;
+		Rasters.SampleType sampleType = null;
 
 		List<Integer> sampleFormat = getSampleFormat();
 		int format = sampleFormat != null && sampleIndex < sampleFormat.size() ? sampleFormat
```

Comment:
check for sampleIndex < sampleFormat.size() is not necessary (tiff is corrupt in this case, i realised that my test generates tiffs that are corrupt in this manner). But please add a similar check for null for getBitsPerSample() which defaults to 1